uPortal-webapp/src/main/resources/properties/jgroups.xml

```diff
@@ -27,66 +27,58 @@
 -->
 <config xmlns="urn:org:jgroups"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-        xsi:schemaLocation="urn:org:jgroups http://www.jgroups.org/schema/JGroups-3.1.xsd">
-    <TCP bind_port="${uPortal.cacheManager.jgroups.tcp.bind_port}"
-         loopback="false"
-         recv_buf_size="${tcp.recv_buf_size:20M}"
-         send_buf_size="${tcp.send_buf_size:640K}"
+        xsi:schemaLocation="urn:org:jgroups http://www.jgroups.org/schema/JGroups-3.6.xsd">
+
+    <!-- bind_addr can be set via environment properties -->
+    <!-- initial_hosts can be set with other nodes. Format = IP[port],... -->
+    <TCP bind_port="${jgroups.tcp.bind_port:7800}"
+         recv_buf_size="${jgroups.tcp.recv_buf_size:20M}"
+         send_buf_size="${jgroups.tcp.send_buf_size:640K}"
          max_bundle_size="64K"
-         max_bundle_timeout="30"
-         use_send_queues="true"
          sock_conn_timeout="300"
 
-         timer_type="new"
-         timer.min_threads="4"
-         timer.max_threads="10"
-         timer.keep_alive_time="3000"
-         timer.queue_max_size="500"
-         
-         thread_pool.enabled="true"
-         thread_pool.min_threads="1"
-         thread_pool.max_threads="10"
-         thread_pool.keep_alive_time="5000"
-         thread_pool.queue_enabled="false"
-         thread_pool.queue_max_size="100"
-         thread_pool.rejection_policy="discard"
-
-         oob_thread_pool.enabled="true"
-         oob_thread_pool.min_threads="1"
-         oob_thread_pool.max_threads="8"
-         oob_thread_pool.keep_alive_time="5000"
-         oob_thread_pool.queue_enabled="false"
-         oob_thread_pool.queue_max_size="100"
-         oob_thread_pool.rejection_policy="discard"/>
+         thread_pool.min_threads="0"
+         thread_pool.max_threads="20"
+         thread_pool.keep_alive_time="30000"
+    />
 
+    <!-- user database for discovery -->
     <JDBC_PING
             connection_url="${hibernate.connection.url}"
```

Comment:
How is this file able to resolve these properties?  It doesn't seem like it can.

Reply:
That file does honor system properties substitution. To make this work, leveraging `SystemPropertySetter` in applicationContext.xml to set system properties from the Spring context.

Reply:
Nice.